### PR TITLE
Consf 72 Adding docs for using this library

### PR DESCRIPTION
**Pull request by lizlux**
This PR adds a /docs folder with information on how to work with the code. 

We can add docs on working with a distribution of this code at a later date, once we have that.

### CODE REVIEW

File: docs/persistence.md [hunk 1]
```diff
@@ -0,0 +1,7 @@
+# Saving an infobox
+
+To save the infobox, simply call:
+```javascript
+infobox.save();
+```
+To save to MediaWiki (which is the default data store), the `InfoboxTemplateBuilder` instance must contain a `persistOptions` property with a `title` value set.
```
Comment (kenkouot) [inline docs/persistence.md]: Technically MW isn't a data store. Just call it persistence target.

File: docs/serializers.md [hunk 2]
```diff
@@ -0,0 +1,9 @@
+# Serialization
+
+Serialization methods live in the [src/serializers](../src/serializers) directory. Each serializer in that directory should contain a `serialize` and `desrialize` method.
```
Comment (kenkouot) [inline docs/serializers.md:3]: Can you provide method signatures here? It'd be more illustrative.

File: docs/api.md [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+// add the second element to the group
+group.add(field2);
+
+// add the group to the infobox data instance
+infobox.data.add(group);
+```
+Note that the `boundVariableName` is the key that identifies a cell in an infobox.  
+
+#### Events
+This app uses an [EventEmitter](https://github.com/Olical/EventEmitter) library, which can be used for communication between user actions and the core API. For more information on the EventEmitter API, see [the docs](https://github.com/Olical/EventEmitter/blob/master/docs/api.md).
```
Comment (kenkouot) [inline docs/api.md]: We want to list the events we make available

**Comment by jkuner** [comment]
What about an overview document, or adding an overview to the main readme?

**Pull request merged**